lesson12/script.js

```diff
@@ -0,0 +1,59 @@
+function showLoadingImg(){
+    const loadingImg = document.createElement('img');
+    loadingImg.src = "loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+async function getData(){
+    showLoadingImg();
+    try{
+        const endpoint = "https://api.json-generator.com/templates/szdgGQcOLXuk/data?access_token=hu4bc7qh9znx2m8f53mn4mz2hryvdntkavwbw8j0";
+        const response = await fetch(endpoint);
+        //responseがokではないとき
+        if(!response.ok){
+        throw new Error(`${response.status}:${response.text}`);
+        }
+        //responseがokのとき
+        const json = await response.json();
+        const data = await json.data;
+        return data;
+    }catch(e){
+        document.getElementById('js-list').textContent = e;
+        console.error(e);
+    }finally{
+        removeLoadingImg();
+    }
+}
+
+async function  renderList(){
+    const result = await getData();
+    //resultがtrueではないとき
+    if (!result) {
+        return;
+    }
```

Comment:
IMO
How about creating a function to get the result and passing it to the renderList as an argument?

Reply:
Thank you for your comment!

I think it's a very good idea!

Is it correct to cut out the next part as a separate function?
````
        const endpoint = "https://api.json-generator.com/templates/szdgGQcOLXuk/data?access_token=hu4bc7qh9znx2m8f53mn4mz2hryvdntkavwbw8j0";
        const response = await fetch(endpoint);
        //responseがokではないとき
        if(!response.ok){
        throw new Error(`${response.status}:${response.text}`);
        }
        //responseがokのとき
        const json = await response.json();
        const data = await json.data;
        return data;
````

Reply:
I think it's a good idea!Let's try it!

Reply:
I will try!
I thought this part was long, so I got an idea from you and it was very helpful!
Thank you!

Reply:
I separeted fetch the data process.
https://github.com/nakagawamai/js-lesson/pull/14/commits/e5efcf1b67107ba43dd00e442c095c49c4860919

Reply:
```suggestion
async function init() {
  showLoadingImg();
  try {
    const result = await fetchdata();
    renderList(result);
  } catch (e) {
    document.getElementById("js-list").textContent = e;
    console.error(e);
  } finally {
    removeLoadingImg();
  }
}

function renderList(result) {
  //resultがtrueではないとき
  if (!result) {
    return;
  }
```
How about separating the work by each function?
Make getData an init and a place to execute.
I will leave it to you to decide if you want to modify it or not.

Reply:
Thank you for your idea!
I wll separating make the list function.

I thought that the following should be left in this place, 
so I cut out only the process of creating a list into a function.
```suggestion
 async function init() {
  showLoadingImg();
  try {
    const result = await fetchdata();
     //resultがtrueではないとき
    if (!result) {
       return;
    }
    //resultがtrueのとき
    renderList(result);
  } catch (e) {
    document.getElementById("js-list").textContent = e;
    console.error(e);
  } finally {
    removeLoadingImg();
  }
}
```